aminator/plugins/cloud/ec2.py

```diff
@@ -358,16 +397,37 @@ def register_image(self, *args, **kwargs):
         if vm_type == 'hvm':
             if context.ami.get("enhanced_networking", False):
                 ami_metadata['sriov_net_support'] = 'simple'
+            if context.ami.get("ena_networking", False):
+                ami_metadata['ena_networking'] = True
 
         if not self._register_image(**ami_metadata):
             return False
+
         return True
 
-    def _make_block_device_map(self, block_device_map, root_block_device):
-        bdm = BlockDeviceMapping(connection=self._connection)
-        bdm[root_block_device] = BlockDeviceType(snapshot_id=self._snapshot.id, delete_on_termination=True)
+    def _make_block_device_map(self, block_device_map, root_block_device, delete_on_termination=True):
+        """ construct boto3 style BlockDeviceMapping """
+
+        bdm = []
+
+        # root device
+        root_mapping = {}
+        root_mapping['DeviceName'] = root_block_device
+        root_mapping['Ebs'] = {}
+        root_mapping['Ebs']['SnapshotId'] = self._snapshot.id
+        root_mapping['Ebs']['VolumeSize'] = self._volume.size
+        root_mapping['Ebs']['VolumeType'] = self._volume.type
+        root_mapping['Ebs']['DeleteOnTermination'] = delete_on_termination
+        bdm.append(root_mapping)
+
+        # ephemerals
         for (os_dev, ec2_dev) in block_device_map:
-            bdm[os_dev] = BlockDeviceType(ephemeral_name=ec2_dev)
+            mapping = {}
+            mapping['VirtualName'] = ec2_dev
+            mapping['DeviceName'] = os_dev
+        bdm.append(mapping)
```

Comment:
Bad indent? If I'm grokking this right, you want a `mapping` dict for each `os_dev` and `ec2_dev` pair in `block_device_map` but if you append outside the for loop, you're only ever going to have a single block device mapping.